packages/kbn-server-route-repository/src/register_routes.ts

```diff
@@ -141,16 +144,18 @@ export function registerRoutes<TDependencies extends Record<string, any>>({
       router[method](
         {
           path: pathname,
-          options,
+          options: omit(options, 'security'),
           validate: validationObject,
+          security: options.security,
```

Comment:
I don't think this is correct because they're different things (see [Slack](https://elastic.slack.com/archives/C9097ABGC/p1732179484087239?thread_ts=1732118911.705509&cid=C9097ABGC)).
I'm making a similar change in a branch but adding security as its own top level property because you may want to use both.

But it raises questions about how much we want to "leak" through the Core APIs, similar for the access setting below.

Reply:
Fixed this now. I've defined `security` as an optional top-level parameter. There's some routes in Streams that still use `options.security`, but I'll make those changes after Elena's fix.

Reply:
Based on [this](https://github.com/elastic/kibana/pull/201352#issuecomment-2493850741), that sounds good to me 👍🏼

Reply:
Elena's PR is merged